analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -259,6 +259,9 @@ node.Parent switch
                 MemberAccessExpressionSyntax memberAccess => node == memberAccess.Name ? ParentAccessType(memberAccess) : SymbolAccess.Read,
                 // _?.node : node?._
                 MemberBindingExpressionSyntax memberBinding => node == memberBinding.Name ? ParentAccessType(memberBinding) : SymbolAccess.Read,
+                // node ??= _ : _ ??= node
+                AssignmentExpressionSyntax assignmentExpression when assignmentExpression.IsKind(SyntaxKindEx.CoalesceAssignmentExpression) =>
```

Comment:
Everything is expression and syntax around here :)
```suggestion
                AssignmentExpressionSyntax assignment when assignment.IsKind(SyntaxKindEx.CoalesceAssignmentExpression) =>
```